addons/iap/models/iap_account.py

```diff
@@ -75,6 +75,15 @@ def get_credits_url(self, service_name, base_url='', credit=0, trial=False):
             d.update({'trial': trial})
         return '%s?%s' % (base_url, werkzeug.urls.url_encode(d))
 
+    @api.model
+    def get_notification_message(self, service_name, notify_type, message=''):
+        if notify_type == 'warning':
+            data = self.env['iap.account'].get_credits_url(service_name, force_create=False)
+            message = _('%s <br/> <a href="%s" target="_blank"> <i class="fa fa-arrow-right"/> IAP Credit</a>') % (message, data)
+        self.env['bus.bus'].sendone(
+            (self._cr.dbname, 'res.partner', self.env.user.partner_id.id),
+            {'type': 'simple_notification', 'message': message, 'sticky': False, 'simple_message': True, notify_type: True})
```

Comment:
We have plans to change or maybe even remove `simple_notification` and/or its support for html content, as well as refactor all the notifications in general.

It would be better if you created a notification specific for this case and handled it manually in JS. Building a template for the JS in python isn't very clean.

Minimum payload should be sent. The only data that should be forwarded to the client for this notification is the original error message and the iap url itself. Then JS client should take care of formatting and displaying it.